### PR TITLE
feat(auth): enhance token handling for API requests

### PR DESCRIPTION
**Pull request by rossnelson**
## Description & motivation 💭 

This PR enhances the authentication handling for API requests by improving token management and integration. Key changes include:
- Adding access tokens as query parameters in `getEndpointForRawHistory` when available.
- Centralizing token retrieval logic in `authTokens` for better maintainability.
- Enhancing `withAuth` to robustly handle both access and ID tokens.
- Updating the `load` function in the history page to await tokenized endpoints.

These changes ensure secure and efficient API requests while simplifying token management.

### Screenshots (if applicable) 📸 

N/A

### Design Considerations 🎨 

- Ensured backward compatibility with existing API requests.
- Verified that token handling adheres to security best practices.

### How was this tested 👻 

- [x] Manual testing to verify token integration in API requests.
- [ ] E2E tests added (pending).
- [ ] Unit tests added (pending).

### Steps for others to test: 🚶🏽‍♂️🚶🏽‍♀️ 

1. Verify that API requests include the correct tokens in query parameters.
2. Test the `load` function in the history page to ensure it handles tokenized endpoints correctly.
3. Validate that `authTokens` centralizes token retrieval without breaking existing functionality.

### Draft Checklist 

- [ ] Complete E2E and unit tests.
- [ ] Review token handling for edge cases.

### Merge Checklist 

- [ ] Ensure all tests pass.
- [ ] Confirm documentation updates are complete.

### Issue(s) closed 

N/A

### Any docs updates needed? 

- Update README to reflect changes in token handling if necessary.

### CODE REVIEW

**Comment by rossnelson** [comment]
We can test this change using the `generate-ui-pr` GitHub action prior to merging this.

File: src/lib/services/events-service.ts [hunk 1]
```diff
@@ -53,6 +55,14 @@ export const getEndpointForRawHistory = ({
     },
     true,
   );
+
+  const url = new URL(route);
+
+  if (!tokens.accessToken) return url.toString();
+
+  url.searchParams.append('accessToken', tokens.accessToken);
```
Comment (Alex-Tideman) [inline src/lib/services/events-service.ts:63]: Does ui-server have the logic to use this search param? Wasn't seeing anything in there

Comment (rossnelson) [reply src/lib/services/events-service.ts:63]: Sorry, I meant to put this back in draft. I am working on that side of things now.
